### ISSUE TITLE
Support custom size bits or just bits in magic signatures

### ISSUE THREAD

**Issue by kalidasya** [kaitai-io/kaitai_struct#578]
Hey, I have a usecase (mpeg-ts parser) where there are magic signatures which are arbitrary long bit sized integers. The following usecase currently is not supported:
```
- id: fixed_10
  type: b2
  contents: [2]
```
I thought this could be a workaround (less readable maybe):
```
- id: fixed_10
  contents: [true, false]
```
but it also does not work. It is not a big deal as it is just for confirmation, but would be nice if its supported.

**Comment by KOLANICH** [comment]
#81 #435

**Comment by GreyCat** [comment]
Indeed, looks like this use case will be completely covered in #435. @kalidasya, please tell me if it will work for you?

**Comment by kalidasya** [comment]
@GreyCat it can cover it, bit size values are not mentioned in that proposal, I will comment it then this can be closed.
